pkg/secrethub/auth_aws.go

```diff
@@ -0,0 +1,62 @@
+package secrethub
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/auth"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+)
+
+type awsAuthService struct {
+	client client
+}
+
+func newAWSAuthService(client client) AuthMethodService {
+	return &awsAuthService{
+		client: client,
+	}
+}
+
+func (s awsAuthService) Authenticate() (auth.Authenticator, error) {
+	// Currently always use the eu-west-1 region.
+	region := endpoints.EuWest1RegionID
+
+	cfg := aws.NewConfig().WithRegion(region).WithEndpoint("sts." + region + ".amazonaws.com")
+	awsSess, err := session.NewSession(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("could not get AWS session: %v", err)
+	}
+
+	svc := sts.New(awsSess, cfg)
+
+	stsReq, _ := svc.GetCallerIdentityRequest(&sts.GetCallerIdentityInput{})
+
+	err = stsReq.Sign()
+	if err != nil {
+		return nil, fmt.Errorf("could not sign STS request: %v", err)
+	}
+
+	var buf bytes.Buffer
+	err = stsReq.HTTPRequest.Write(&buf)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Consider moving this sts prepare request logic  to a separate helper function to auto document the behavior.